src/core/qgssqliteexpressioncompiler.h

```diff
@@ -13,18 +13,30 @@
  *                                                                         *
  ***************************************************************************/
 
-#ifndef QGSSPATIALITEEXPRESSIONCOMPILER_H
-#define QGSSPATIALITEEXPRESSIONCOMPILER_H
+#ifndef QGSSQLITEEXPRESSIONCOMPILER_H
+#define QGSSQLITEEXPRESSIONCOMPILER_H
 
 #include "qgssqlexpressioncompiler.h"
 #include "qgsexpression.h"
-#include "qgsspatialitefeatureiterator.h"
 
-class QgsSpatiaLiteExpressionCompiler : public QgsSqlExpressionCompiler
+/** \ingroup core
```

Comment:
Can you wrap this whole header (from the includes down) in a 

``` \\\ @cond PRIVATE 
...
\\\ @endcond
```

block? That will prevent the class from even showing in the generated API docs. You'll also need to wrap the .cpp in the same @cond / @endcond block or doxygen will complain.